src/download_boot_artifacts/download_boot_artifacts.go

```diff
@@ -0,0 +1,120 @@
+package download_boot_artifacts
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"path"
+	"syscall"
+
+	"github.com/openshift/assisted-service/models"
+	log "github.com/sirupsen/logrus"
+)
+
+var (
+	hostArtifactsFolder  string = "/host/boot/discovery"
+	artifactsFolder      string = "/boot/discovery"
+	kernelFile           string = "vmlinuz"
+	initrdFile           string = "initrd"
+	bootLoaderFolder     string = "/host/boot/loader/entries"
+	bootLoaderConfigFile string = path.Join(bootLoaderFolder, "/00-assisted-discovery.conf")
+)
+
+const bootLoaderConfigTemplate = `title Assisted Installer Discovery
+version 999
+options random.trust_cpu=on ignition.firstboot ignition.platform.id=metal 'coreos.live.rootfs_url=%s'
+linux %s
+initrd %s`
+
+func Run(infraEnvId, downloaderRequestStr string) error {
+	var req models.DownloadBootArtifactsRequest
+	if err := json.Unmarshal([]byte(downloaderRequestStr), &req); err != nil {
+		log.WithError(err).Errorf("DownloadBootArtifacts: json.Unmarshal")
+		return err
+	}
```

Comment:
I think it's best that if instead of logging at every error site, we wrap the error using `fmt.Errorf` and return that, then log higher up the stack.

e.g.:
```suggestion
	if err := json.Unmarshal([]byte(downloaderRequestStr), &req); err != nil {
		return fmt.Errorf("failed to unmarshal request: %w", err)
	}
```

This way both the error that gets logged and the error that gets returned contain the full context information

Reply:
Good idea! Modified to wrap error messages